helper/changelog_generator.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+import os
+import subprocess
+import sys
+
+# Date since then we want to ask the git history for data
+since_date = None
+
+# List of git history log entries
+log_data = []
+
+# Author and Date
+log_author = None
+log_date = None
+
+# changelog header line
+log_start = '-' * 67 + os.linesep
+
+# commit message
+commit_message = []
+
+# Open reference changelog file
+# This assumes a certain changelog format
+with open('package/python-kiwi.changes', 'r') as changelog:
+    # read headline of first entry
+    changelog.readline()
+    # read date of first entry
+    since_date = changelog.readline().split('-')[0]
+
+# Read git history since latest entry from reference log
+process = subprocess.Popen(
+    [
+        'git', 'log', '--topo-order', '--no-merges',
+        '--since="{0}"'.format(since_date)
+    ],
+    stdout=subprocess.PIPE
+)
+
+# Iterate over history and convert to changelog format
+for line in iter(process.stdout.readline, b''):
+    if line.startswith('commit'):
+        if commit_message:
+            log_data.append(log_start)
+            log_data.append(
+                '{0} - {1}{2}{2}'.format(
+                    log_date, log_author, os.linesep
+                )
+            )
+            commit_message.pop(0)
+            log_data.append('- {0}{1}'.format(
+                commit_message.pop(0).lstrip(), os.linesep)
+            )
+            for line in commit_message:
+                log_data.append('  {0}{1}'.format(
+                    line.lstrip(), os.linesep)
+                )
+            commit_message = []
+    elif line.startswith('Author:'):
+        log_author = line.replace('Author:', '').strip()
+    elif line.startswith('Date:'):
+        log_date = line.replace('Date:', '').strip()
```

Comment:
This date needs to be converted to the following format: `%a %b %d %T %Z %Y`

Reply:
done using git log to do the job